src/json/def_to_json.cr

```diff
@@ -0,0 +1,186 @@
+module JSON
+  # The `def_to_json` macro generates a `to_json(json : JSON::Builder)` method based on provided *mappings*.
+  #
+  # It is a lightweight alternative to `JSON.mapping` if you don't need to declare instance variables and a parser.
```

Comment:
I'd phrase this in terms of "serialization-only" or "read-only" instead of "if you don't need to declare instance variables and a parser" because your docs don't get near the heart of what `def_to_json` is used for.